iphone/cli/commands/_build.js

```diff
@@ -4902,1224 +5021,1247 @@ iOSBuilder.prototype.writeDebugProfilePlists = function writeDebugProfilePlists(
-						logger: this.logger,
-						builder: this,
-						jsFiles,
-						jsBootstrapFiles,
-						sdkCommonFolder,
-						defaultAnalyzeOptions: {
-							minify: this.minifyJS,
-							transpile: this.transpile,
-							sourceMap: this.sourceMaps,
-							resourcesDir: this.xcodeAppDir,
-							logger: this.logger,
-							targets: {
-								ios: this.minIosVersion
-							}
-						}
-					});
-					task.run()
-						.then(() => {
-							this.tiSymbols = task.data.tiSymbols;
+			if (launchLogo) {
+				this.logger.info('  ' + desc);
+			}
+		});
+	}
 
-							return next(); // eslint-disable-line promise/no-callback-in-promise
-						})
-						.catch(e => {
-							this.logger.error(e);
-							process.exit(1);
-						});
-				},
+	await this.writeAssetContentsFile(path.join(assetCatalogDir, 'Contents.json'), {
+		images: images,
+		info: {
+			version: 1,
+			author: 'xcode'
+		}
+	});
 
-				function writeBootstrapJson() {
-					this.logger.info(__('Writing bootstrap json'));
+	// If there are no missing launch logos, we're done
+	if (!missingLaunchLogos.length) {
+		return [];
+	}
 
-					const originalBootstrapJsonName = path.join('ti.internal', 'bootstrap.json');
-					const bootstrapJsonRelativePath = this.encryptJS ? path.join('ti_internal', 'bootstrap_json') : originalBootstrapJsonName;
-					const bootstrapJsonAbsolutePath = path.join(this.encryptJS ? this.buildAssetsDir : this.xcodeAppDir, bootstrapJsonRelativePath);
-					const bootstrapJsonString = JSON.stringify({ scripts: jsBootstrapFiles });
+	if (!this.forceRebuild) {
+		this.logger.info(__('Forcing rebuild: launch logos changed since last build'));
+		this.forceRebuild = true;
+	}
 
-					if (this.encryptJS) {
-						this.jsFilesEncrypted.push(originalBootstrapJsonName); // original name
-						this.jsFilesToEncrypt.push(bootstrapJsonRelativePath); // encrypted name
-					}
+	if (!this.buildOnly && (this.target === 'device' || this.target === 'simulator')) {
+		this.logger.warn(__('If this app has been previously installed on this %s, you may need restart it to see the latest launch logo', this.target));
+		this.logger.warn(__('iOS renders and caches the launch screen to a PNG image that seems to only be invalidated by restarting iOS'));
+	}
 
-					if (!fs.existsSync(bootstrapJsonAbsolutePath) || (bootstrapJsonString !== fs.readFileSync(bootstrapJsonAbsolutePath).toString())) {
-						this.logger.debug(__('Writing %s', bootstrapJsonAbsolutePath.cyan));
+	if (!launchLogo) {
+		// just use the DefaultIcon.png to generate the missing LaunchLogos
+		return missingLaunchLogos;
+	}
 
-						fs.ensureDirSync(path.dirname(bootstrapJsonAbsolutePath));
-						fs.writeFileSync(bootstrapJsonAbsolutePath, bootstrapJsonString);
-					} else {
-						this.logger.trace(__('No change, skipping %s', bootstrapJsonAbsolutePath.cyan));
-					}
+	// Resize the launch logo to the other dimensions needed
+	await new Promise((resolve, reject) => {
```

Comment:
<!--
  0 failure: 
  1 warning:  iphone/cli/comman...
  
  
  DangerID: danger-id-Danger;
  File: iphone/cli/commands/_build.js;
  Line: 6093;
-->

- :warning: iphone/cli/commands/_build.js line 6093 – 'reject' is defined but never used. Allowed unused args must match /^_.+/u. (no-unused-vars)